### PR TITLE
fix: build vm strategy for the arm machine

### LINKED ISSUE #1397: vm build strategy not working on mac m1
**Describe the bug**
building wrapper with vm build strategy fails on m1 mac with following error:
```
qemu: uncaught target signal 11 (Segmentation fault) - core dumped

error:
 
could not compile `syn`



Caused by:

  process didn't exit successfully: `rustc --crate-name syn --edition=2018 /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.103/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debug-assertions=off --cfg 'feature="clone-impls"' --cfg 'feature="default"' --cfg 'feature="derive"' --cfg 'feature="parsing"' --cfg 'feature="printing"' --cfg 'feature="proc-macro"' --cfg 'feature="quote"' -C metadata=0bc87bee3b0d6ea7 -C extra-filename=-0bc87bee3b0d6ea7 --out-dir /project/target/release/deps -L dependency=/project/target/release/deps --extern proc_macro2=/project/target/release/deps/libproc_macro2-e0d2cd6cdab0f8d4.rmeta --extern quote=/project/target/release/deps/libquote-0e5aeed9eb5411d4.rmeta --extern unicode_ident=/project/target/release/deps/libunicode_ident-8336050b0a2aef5b.rmeta --cap-lints allow --cfg syn_disable_nightly_tests` (signal: 11, SIGSEGV: invalid memory reference)
```

**Please provide the steps to reproduce and if possible a minimal demo of the problem.**
Run `polywrap build` on any mac m1 machine

**Expected behavior**
It should build the wrapper

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
- Closes: #1397

## Changelog
- feat: docker image uses alpine linux as base image
- fix: VM build strategy now uses image with appropriate architecture to build the wrapper

### CODE REVIEW

**Review by dOrgJelli — DISMISSED** [review]
LGTM!

**Review by krisbitney — CHANGES_REQUESTED** [review]
I still get this:
Screenshot 2023-01-18 at 4 10 14 PM

**Review by cbrzn — DISMISSED** [review]
edit: Niraj fixed the latest version in docker so we good :-)

**Review by krisbitney — APPROVED** [review]

**Comment by Niraj-Kamdar** [comment]
> I still get this: Screenshot 2023-01-18 at 4 10 14 PM

Turned out this was the issue on @krisbitney side so all good

**Review by dOrgJelli — APPROVED** [review]

File: packages/cli/src/lib/defaults/build-strategies/wasm/rust/vm/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 
 # Install the wasm32 rust build target
 RUN rustup target add wasm32-unknown-unknown
 
 WORKDIR /build-deps
 
 # Install curl
-RUN apt-get update
-RUN apt-get -y install curl clang llvm build-essential
+RUN apk add curl build-base pkgconfig openssl-dev bash
```
Comment (dOrgJelli) [inline packages/cli/src/lib/defaults/build-strategies/wasm/rust/vm/Dockerfile:9]: Hey @Niraj-Kamdar, apologies for not asking about this before, but why are we now installing `build-base`, `pkgconfig`, `openssl-dev` and `base`?

Also I should have caught this before, but for rust builds we require `clang`, `llvm` and `build-essentials` to build & link native C libs.

File: packages/cli/src/lib/defaults/build-strategies/wasm/rust/image/Dockerfile.mustache [hunk 2]
```diff
@@ -1,13 +1,12 @@
-FROM rust:1.65.0 as base
+FROM rust:1.66-alpine as base
```
Comment (dOrgJelli) [inline packages/cli/src/lib/defaults/build-strategies/wasm/rust/image/Dockerfile.mustache:1]: Also what is the reasoning for moving to alpine?

File: packages/cli/src/lib/defaults/build-strategies/wasm/rust/vm/Dockerfile (hunk 1 above)
Comment (dOrgJelli) [reply packages/cli/src/lib/defaults/build-strategies/wasm/rust/vm/Dockerfile:9]: One answer I found is that `build-base` is the alpine equivalent of `build-essentials`.

**Pull request merged**